### ISSUE TITLE
[Bug]: Shared color, high contrast slot assignment update

### ISSUE THREAD

**Issue by daisygeng** [microsoft/fluentui#24256]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://aka.ms/fluentui-storybook
```

### Reproduction

https://aka.ms/fluentui-storybook

### Bug Description

[Shared color] BG 1 from **hcCanvasText -> hcCanvas**
![image](https://user-images.githubusercontent.com/48726002/183119815-d8eea3a3-a79d-4113-8a76-d046c8abd32b.png)
![image](https://user-images.githubusercontent.com/48726002/183120005-98e1e7ec-c21a-4222-99d4-854450beb1d5.png)
![image](https://user-images.githubusercontent.com/48726002/183120054-56e50635-1aab-4659-9996-d058d0f5b492.png)
![image](https://user-images.githubusercontent.com/48726002/183120091-6592c466-67e8-46d4-a63e-2c5df5a7683f.png)
![image](https://user-images.githubusercontent.com/48726002/183120155-229136ef-0a02-4311-8a6a-e6353f7664fd.png)

[Shared color] FG 1 from **hcCanvas -> hcCanvasText**
![image](https://user-images.githubusercontent.com/48726002/183119830-31bf8f38-cf6c-4c8e-b742-a619960bf054.png)
![image](https://user-images.githubusercontent.com/48726002/183120037-c0afd807-277d-45ac-a022-a19e24ad7df9.png)
![image](https://user-images.githubusercontent.com/48726002/183120067-294d4f48-2bfb-43c9-b844-47f3638de5ec.png)
![image](https://user-images.githubusercontent.com/48726002/183120114-803674c9-dcd2-4e3b-bc00-a9dc37009184.png)
![image](https://user-images.githubusercontent.com/48726002/183120169-76859306-41cb-4f4e-906e-07c547d087d1.png)

------
(For Yellow FG 1 only, change should not impact other shared colors)
Yellow FG 1 from **Yellow Shade 10 -> Yellow Shade 30**
![image](https://user-images.githubusercontent.com/48726002/183120438-6c3969b3-9091-4446-b472-ee98831bfe6c.png)

------

Badge Tint spec updated to reflect update
![image](https://user-images.githubusercontent.com/48726002/183120698-619739c5-ac98-4452-bf89-bb1bcc0ba546.png)

-----

This update will help resolve this Teams bug:
![image](https://user-images.githubusercontent.com/48726002/183120508-bbc64939-133d-46a4-b3bb-c1b99ffd1ca2.png)

error string not visible in high contrast. This is because this element is using the same alias color token that is assigned to the tint badge. (Red Foreground 1)

Red Foreground 1 currently has Canvas (black) assigned in high contrast directly onto the content background, so in high contrast theme today the text is black on black. 

Tint badge has been fine because it's paired w/ the background (Red Background 1) so as long as we change both FG1 and BG1 at once the badge should be unharmed. 
![image](https://user-images.githubusercontent.com/48726002/183120536-d900e6fa-d440-4820-b4a5-98fea73a1e5b.png)

------
cc @miroslavstastny

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by JustSlone** [comment]
@daisygeng can you clarify the before/after change here. For FG1 it says hcCanvasText -> hcCanvas but the image shows hcCanvasText.

**Comment by daisygeng** [comment]
Typo on my end. Edited original message. FG1 should be hcCanvasText
